app/bundles/UserBundle/Tests/Functional/UserPreferenceTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Tests\Functional;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\UserBundle\Entity\Role;
+use Mautic\UserBundle\Entity\User;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\PasswordHasherInterface;
+
+/**
+ * Functional tests for user preference functionality.
+ *
+ * This test class verifies that:
+ * - Users can successfully set their preferences via API endpoint
+ * - Preferences are properly stored in the database
+ * - The system responds with appropriate success messages
+ *
+ * Specifically tests the column visibility preference for contacts as a representative example.
+ */
+class UserPreferenceTest extends MauticMysqlTestCase
+{
+    public function testUserCanSetPreference(): void
+    {
+        $user = $this->createTestUser();
+
+        $this->loginUser($user);
+        $this->client->setServerParameter('PHP_AUTH_USER', $user->getUsername());
+        $this->client->setServerParameter('PHP_AUTH_PW', 'Maut1cR0cks!');
```

Comment:
Is this necessary? The admin user is already created and logged in for every functional test.